walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -36,16 +36,34 @@ public int compare(Event a, Event b) {
       };
 
   /**
-   * Returns a list of time periods in which the meeting, specified by request, could happen.
+   * Returns a list of time periods in which the meeting, specified by request, could happen. If one
+   * or more time slots exists so that both mandatory and optional attendees can attend, it returns
+   * those time slots. Otherwise, it returns the time slots that fit just the mandatory attendees.
    *
    * @param eventsCollection the events we know about
    * @param request information about the meeting, including attendees, optional attendees, and how
    *     long it needs to be
    */
   public Collection<TimeRange> query(Collection<Event> eventsCollection, MeetingRequest request) {
-    ArrayList<Event> events =
-        getRelevantEvents(
-            new HashSet<String>(request.getAttendees()), new ArrayList<Event>(eventsCollection));
+    Collection<TimeRange> withOptionalAttendees =
+        getMeetingTimes(eventsCollection, request, /*includeOptionalAttendees=*/ true);
+
+    // Special case: if no mandatory attendees and optional attendees' schedules cannot fit in a
+    // meeting, no meeting times are possible.
+    return !withOptionalAttendees.isEmpty() || request.getAttendees().isEmpty()
```

Comment:
I find this combination of non-trivial boolean expression plus ternary hard to quickly parse.

My preference

```java
if (!withOptionalAttendees.isEmpty()) {
  return withOptionalAttendees;
}

return getMeetingTimes(eventsCollection, request, /*includeOptionalAttendees=*/ false);
```

This potentially does a computation of meeting times with zero requested attendees, but I expect that to be cheap.